### PR TITLE
Use `check-run-results` for `run-fail` test stderr

### LINKED ISSUE #65865: Cleanup: Migrate run-fail tests in src/test/ui from error-pattern to check-run-results
As tests are moved from `src/test/run-fail` to `src/test/ui` #65440 the meaning
of `// error-pattern` in ui tests becomes overloaded , since tests in
`src/test/run-fail` match error patterns to executable output while
`src/test/ui` tests match them to compiler output.

The changes in #65759 enable such overloaded use: if test is expected to
compile successfully and will be run, the patterns are matched to executable
output, otherwise patterns are matched to the compiler output.

@petrochenkov [suggested](https://github.com/rust-lang/rust/pull/65759#discussion_r339302684) that it might be preferable to migrate those test to use check-run-result instead.

1. [ ] Decide if check-run-result is preferable over error-pattern for run-fail tests in ui.
2. [ ] Migrate run fail tests to use check-run-result.
3. [ ] Remove support for overloaded meaning of error-pattern.

### PR DESCRIPTION
**Pull request by clubby789**
Fixes #65865

### CODE REVIEW

File: src/tools/tidy/src/ui_tests.rs [hunk 1]
```diff
@@ -17,7 +17,7 @@ use ignore::Walk;
 const ENTRY_LIMIT: u32 = 901;
 // FIXME: The following limits should be reduced eventually.
 
-const ISSUES_ENTRY_LIMIT: u32 = 1672;
```
Comment (clubby789) [inline src/tools/tidy/src/ui_tests.rs:20]: This increased as there's now some extra output files for existing issue tests

**Comment by clubby789** [comment]
oops, non anonymized panic strings

**Comment by clubby789** [comment]
Reverted a few tests that have more variable output

**Comment by clubby789** [comment]
Hmm, we can't normalize `$CARGO_HOME` naively like other paths, because we have tests with paths containing `.../cargo...` - and `/cargo` is the CI `$CARGO_HOME`

**Comment by clubby789** [comment]
Dropped the cargo home normalization since I think it's going to be a headache to deal with, especially as line numbers change in external crates

**Comment by estebank** [comment]
My apologies, I should have approved before the last merge conflict. r=me after fixing that.

**Comment by clubby789** [comment]
Looks spurious

@bors retry

**Comment by clubby789** [comment]
@bors r-
Misread the error, sanitizer tests are fragile due to addresses and should not have stderr checked

**Comment by jieyouxu** [comment]
Also leaving a comment here so discussions don't get lost elsewhere:

We should be a bit careful with replacing `//@ error-pattern` with `//@ check-run-results`, because they are not mutually exclusive and they aren't really substitutes:

- `//@ error-pattern` will look for a regex match in the run stderr (and stdout combined, IIRC) if the test is `//@ run-fail`.
- `//@ check-run-results` instructs compiletest to snapshot run stderr/stdout, and compare a run's stderr/stdout against those *exactly*.

Concretely, this means:

- `//@ check-run-results` is for when you *really* want the *exact* run stderr/stdout to not change. This will be especially relevant to tests that may include things that change between runs.
    - If a test needs such *exact* snapshot comparisons, you can *still* have `error-pattern` to enforce the check on the pattern that is critical, to prevent you from accidentally blessing the important thing away.
- `//@ error-pattern` is good for checking the *critical* parts of the run stderr/stdout, cannot be easily blessed away, and is typically less sensitive than full stderr/stdout comparisons.

**Comment by clubby789** [comment]
Going to close; I think a blanket approach is unhelpful as it's going to be flaky (especially when stderr contains line numbers from external crates or stdlib), and doesn't provide *much* value

**Comment by jieyouxu** [comment]
cc #134888

**Comment by jieyouxu** [comment]
cc [Audit run-fail ui tests to see if they need *exact* run stderr/stdout #134889](https://github.com/rust-lang/rust/issues/134889).

**Pull request closed without merging**